domain/servicefactory/service.go

```diff
@@ -17,12 +18,20 @@ type ServiceFactory struct {
 // NewServiceFactory returns a new service factory which can be used to
 // get new services from.
 func NewServiceFactory(
-	controllerDB, modelDB changestream.WatchableDBFactory,
+	modelUUID model.UUID,
```

Comment:
This doesn't feel nice. Is there any way we can maybe attach the model uuid to the `WatchableDBFactory`, so it becomes a `ModelWatchableDBFactory`, then peel that off from there?

Reply:
I've done some magic around this. Check it out.